resolver_test.go

```diff
@@ -142,3 +142,25 @@ func (s) TestResolverAddressesToEndpoints(t *testing.T) {
 		t.Fatalf("timed out waiting for endpoints")
 	}
 }
+
+// TestResolverAddressesWithTypedNilAttribute ensures no panic if typed-nil attributes within resolver.State.Addresses
+func (s) TestResolverAddressesWithTypedNilAttribute(t *testing.T) {
+	const scheme = "testresolveraddresseswithtypednilattribute"
+	r := manual.NewBuilderWithScheme(scheme)
```

Comment:
Nit: Instead of a const scheme here, you could simply do: `r := manual.NewBuilderWithScheme(t.Name())`

Reply:
done. Should also simply const scheme in `TestResolverAddressesToEndpoints` or next time?